applicationconfig/client/src/main/java/com/azure/applicationconfig/ConfigurationClientCredentials.java

```diff
@@ -0,0 +1,136 @@
+package com.azure.applicationconfig;
+
+import com.azure.common.credentials.AsyncServiceClientCredentials;
+import com.azure.common.http.HttpHeaders;
+import com.azure.common.http.HttpRequest;
+import reactor.core.publisher.Mono;
+
+import javax.crypto.Mac;
+import javax.crypto.spec.SecretKeySpec;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.security.InvalidKeyException;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.Locale;
+import java.util.stream.Collectors;
+
+import static com.azure.applicationconfig.ConfigurationCredentialsPolicy.CONTENT_HASH_HEADER;
+import static com.azure.applicationconfig.ConfigurationCredentialsPolicy.DATE_HEADER;
+import static com.azure.applicationconfig.ConfigurationCredentialsPolicy.HOST_HEADER;
+
+public class ConfigurationClientCredentials implements AsyncServiceClientCredentials {
+    private final CredentialInformation credentials;
+    private final AuthorizationHeaderProvider headerProvider;
+
+    /**
+     *
+     * @param connectionString connection string in the format "Endpoint=_endpoint_;Id=_id_;Secret=_secret_"
+     */
+    public ConfigurationClientCredentials(String connectionString) {
+        credentials = new CredentialInformation(connectionString);
+        headerProvider = new AuthorizationHeaderProvider(credentials);
+    }
+
+    URL baseUri() { return this.credentials.baseUri; }
+
+    @Override
+    public Mono<String> authorizationHeaderValueAsync(HttpRequest request) {
```

Comment:
now I understand why we need `authorizationHeaderValueAsync(HttpRequest request)` instead of `authorizationHeaderValueAsync(String uri)`, sounds good